### PR TITLE
7.0.0

### PR DESCRIPTION
**Pull request by tptee**
# Infrastructure
- Simplify config
- Replace Karma with Mocha + nyc + babel-plugin-istanbul
- Update travis.yml with new coverage command
- Rewrite the test suite to actually use stubs/spies
- Added coverage for `<Fragment>`

# Patch
- Fix a bug where `<Fragment>` ignored route precedence and would mistake concrete routes for parameters (e.g.`<Fragment forRoute='/home' />` would display for routes like `/:wantsparams`) 
- Remove unused code for generating a full URL

# Breaking changes
- Remove the middleware and wrap dispatch in the store enhancer instead
- Rename `initialStateForSSR` to `initialRouterState` and make it private
- The store enhancer now handles initial location state when passed a `pathname` and/or a `query` object.
- Provide a default history instance and remove need to provide a custom instance to both the store enhancer and the provider. Accept a `basename` option in the enhancer for `useBasename`. Accept a `forServerRender` flag to determine which history instance to use (memory or browser).

### CODE REVIEW

**Comment by ryan-roemer** [comment]
Any plans for functional tests for the supported browser matrix now that we're completely simulating unit tests w jsdom?

Also, what's the word on the street of jsdom vs cheerio? Last I heard everyone was moving to cheerio. (Though haven't checked up in a while...)

**Comment by tptee** [comment]
Yep, functional tests will be on the agenda! I'm also working on an example project that we can reuse for those tests.

I'm still fuzzy on the differences between jsdom and cheerio. AFAICT, jsdom provides a headless environment + HTML parser, while cheerio provides an HTML parser plus a jQuery-equivalent API. Enzyme uses cheerio under the hood, so maybe we're OK?

**Comment by tptee** [comment]
Alrighty, this is ready for review!

There's a couple of `sinon` smells in the test suite which I'll address in a later PR.

File: test/.eslintrc [hunk 1]
```diff
@@ -0,0 +1,9 @@
+{
+  "extends": "../.eslintrc",
+  "env": {
+    "mocha": true
+  },
+  "rules": {
+    "no-unused-expressions": "off"
+  }
+}
```
Comment (divmain) [inline test/.eslintrc]: Any reason this is JSON, whereas the base `.eslintrc` is YAML?

File: test/.setup.js [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+var exposedProperties = ['window', 'navigator', 'document'];
+
+global.document = jsdom('');
+global.window = document.defaultView;
+Object.keys(document.defaultView).forEach((property) => {
+  if (typeof global[property] === 'undefined') {
+    exposedProperties.push(property);
+    global[property] = document.defaultView[property];
+  }
+});
```
Comment (divmain) [inline test/.setup.js]: In many ways, I'd prefer to construct a fresh DOM environment for each test.  However, this should okay so long as precautions are taken not to pollute global state within tests.

**Comment by divmain** [comment]
LGTM!

**Comment by baer** [comment]
this PR LGTM but it could use some docs.

**Comment by tptee** [comment]
Ask and ye shall receive! https://github.com/FormidableLabs/redux-little-router/pull/22

**Pull request merged**